monkey/tests/unit_tests/infection_monkey/exploit/powershell_utils/test_utils.py

```diff
@@ -0,0 +1,80 @@
+from infection_monkey.exploit.powershell_utils import utils
+from infection_monkey.model.host import VictimHost
+
+TEST_USERS = ["user1", "user2"]
+TEST_PASSWORDS = ["p1", "p2"]
+
+
+def test_get_credentials__empty_windows_true():
+    credentials = utils.get_credentials([], [], True)
+
+    assert len(credentials) == 1
+    assert credentials[0] == (None, None)
+
+
+def test_get_credentials__empty_windows_false():
+    credentials = utils.get_credentials([], [], False)
+
+    assert len(credentials) == 0
+
+
+def test_get_credentials__username_only_windows_false():
+    credentials = utils.get_credentials(TEST_USERS, [], False)
+
+    assert len(credentials) == 2
+    assert (TEST_USERS[0], "") in credentials
+    assert (TEST_USERS[1], "") in credentials
+
+
+def test_get_credentials__username_only_windows_true():
+    credentials = utils.get_credentials(TEST_USERS, [], True)
+
+    assert len(credentials) == 5
+    assert (TEST_USERS[0], "") in credentials
+    assert (TEST_USERS[1], "") in credentials
+    assert (TEST_USERS[0], None) in credentials
+    assert (TEST_USERS[1], None) in credentials
+
```

Comment:
Let's switch the order of these two functions so we maintain the "windows_true" followed by "windows_false" pattern.